executor/calibrate_resource.go

```diff
@@ -99,41 +174,121 @@ func (e *calibrateResourceExec) Next(ctx context.Context, req *chunk.Chunk) erro
 
 	exec := e.ctx.(sqlexec.RestrictedSQLExecutor)
 	ctx = kv.WithInternalSourceType(ctx, kv.InternalTxnOthers)
+	if len(e.optionList) > 0 {
+		startTime, endTime, err := e.checkDynamicCalibrateOptions()
+		if err != nil {
+			return err
+		}
+		totalKVCPUQuota, err := getTiKVTotalCPUQuota(ctx, exec)
+		logutil.BgLogger().Info("checkDynamicCalibrateOptions", zap.Float64("totalKVCPUQuota", totalKVCPUQuota), zap.Error(err))
+		if err != nil {
+			return err
+		}
+		totalTiDBCPU, err := getTiDBTotalCPUQuota(ctx, exec)
+		logutil.BgLogger().Info("checkDynamicCalibrateOptions", zap.Float64("totalTiDBCPU", totalTiDBCPU), zap.Error(err))
+		if err != nil {
+			return err
+		}
+		rus, err := getRUPerSec(ctx, exec, startTime, endTime)
+		logutil.BgLogger().Info("checkDynamicCalibrateOptions", zap.Any("rus", rus), zap.Error(err))
+		if err != nil {
+			return err
+		}
+		tikvCPUs, err := getTiKVCPUUsagePerSec(ctx, exec, startTime, endTime)
+		logutil.BgLogger().Info("checkDynamicCalibrateOptions", zap.Any("tikvCPUs", tikvCPUs), zap.Error(err))
+		if err != nil {
+			return err
+		}
+		tidbCPUs, err := getTiDBCPUUsagePerSec(ctx, exec, startTime, endTime)
+		logutil.BgLogger().Info("checkDynamicCalibrateOptions", zap.Any("tidbCPUs", tidbCPUs), zap.Error(err))
+		if err != nil {
+			return err
+		}
+		quotas := make([]float64, 0)
+		lowCount := 0
+		tidbCPULowCount := 0
+		tikvCPULowCOunt := 0
+		for idx, ru := range rus {
+			if idx >= len(tikvCPUs) || idx >= len(tidbCPUs) {
+				break
+			}
+			tikvQuota := totalKVCPUQuota / tikvCPUs[idx]
```

Comment:
What if `tikvCPUs[idx] == 0` here, I think check `tikvCPUs[idx]/totalKVCPUQuota` as the cpu usage percentage is a more ergonomic way